packages/nimbus/src/components/checkbox/checkbox.mdx

```diff
@@ -1,78 +1,312 @@
+  logically.
+- **Visual Feedback:** Provide clear visual feedback when the checkbox is
+  checked or unchecked.
+- **Error Handling:** If a checkbox is required, provide clear error messages if
+  the user forgets to check it.
+
+### Usage
+
+- **Single Checkbox:** Used for binary choices (e.g., "I agree to the terms and
+  conditions").
+- **Checkbox Groups:** Allow users to select multiple options from a list.
+- **Within Tables or Lists:** Checkboxes can be used for bulk actions, like
+  selecting multiple items in a table.
+
+### Logical grouping
+
+Maintain a clear and predictable order for checkboxes to facilitate efficient
+scanning and selection.
+
+> [!TIP]\
+> **Do**
+>
+> - Capitalize the first letter of the label.
+> - Use short and clear labeling.
+> - Group information easily for the user to make the correct selection to
+>   accomplish their goals.
 
 ```jsx-live
-const App = () => <Checkbox isIndeterminate>Indeterminate Checkbox</Checkbox>
+const App = () => (
+  <Grid templateColumns="repeat(2, 1fr)" gap="400">
+    <Grid.Item>
+      <Checkbox isIndeterminate>All general fields</Checkbox>
+    </Grid.Item>
+    <Grid.Item>
+      <Stack direction="column">
+        <Checkbox isSelected isDisabled>Product key</Checkbox>
+        <Checkbox isSelected>Name</Checkbox>
+        <Checkbox>Product type key</Checkbox>
+        <Checkbox>Description</Checkbox>
+        <Checkbox>Category keys</Checkbox>
+      </Stack>
+    </Grid.Item>
+  </Grid>
+)
 ```
 
-## Props
+> [!CAUTION]\
+> **Don't**
+>
+> - Don’t overwhelm the user with options that are uncategorized.
+> - There is no reason to use all caps or punctuation.
+
+```jsx-live
+const App = () => (
+  <Grid templateColumns="repeat(2, 1fr)" gap="400">
+    <Grid.Item>
+      <Stack direction="column">
+        <Checkbox isSelected>Product key</Checkbox>
+        <Checkbox isSelected>NAME...</Checkbox>
+        <Checkbox>Valid from</Checkbox>
+        <Checkbox>Discount prices</Checkbox>
+        <Checkbox>Asset name!</Checkbox>
+      </Stack>
+    </Grid.Item>
+    <Grid.Item>
+      <Stack direction="column">
+        <Checkbox isSelected>Price key?</Checkbox>
+        <Checkbox isSelected>Price value.</Checkbox>
+        <Checkbox>PRICE COUNTRY</Checkbox>
+        <Checkbox isSelected>Valid until</Checkbox>
+        <Checkbox>Category keys.</Checkbox>
+      </Stack>
+    </Grid.Item>
+  </Grid>
+)
+```
+
+### Positive framing
+
+Frame checkbox actions with affirmative language to promote clarity and user
+confidence.
+
+> [!TIP]\
+> **Do**
+>
+> - Simple labels provide clear options for users.
+
+```jsx-live
+const App = () => (
+  <Stack direction="row" gap="1400" alignItems="center">
+    <Text fontSize="300" fontWeight="500">Preferences</Checkbox>
+    <Checkbox isSelected>Include product and variant information</Checkbox>
+  </Stack>
+)
+```
+
+> [!CAUTION]\
+> **Don't**
+>
+> - Don’t contradict available options. This information should be simplified
```

Comment:
ND: This doesn't display well for a single list, but no worries, I think @jaikamat already handled that in one of the PRs. 

<img width="812" alt="image" src="https://github.com/user-attachments/assets/c40c0d47-5402-489d-ac28-9c48acc84dab" />